pywbem_mock/_inmemoryrepository.py

```diff
@@ -0,0 +1,431 @@
+        """
+        Overrides the abstract method.
+
+        Creates a new cim_object in the object store dictionary or fails
+        with an exception if the name already exists in the dictionary.
+        """
+
+        assert isinstance(cim_object, self._cim_object_type)
+
+        if name in self._data:
+            raise ValueError('{} already in {} object store'
+                             .format(name, self._cim_object_type))
+
+        self._data[name] = cim_object.copy()
+
+    def update(self, name, cim_object):
+        """
+        Overrides the abstract method.
+
+        This method updates the corresponding dictionary with the cim_object
+        if the object exists in the dictionary. Otherwise it returns
+        an exception.
+
+        """
+
+        assert isinstance(cim_object, self._cim_object_type)
+
+        if name not in self._data:
+            raise KeyError('{} not in {} object store'
+                           .format(name, self._cim_object_type))
+
+        # Replace the existing object with a copy of the input object
+        self._data[name] = cim_object.copy()
+
+    def delete(self, name):
+        """
+        Overrides the abstract method.
+
+        If the object exists in the object store dictionary, it is deleted.
+        If it does not exist an exception is returned.
+        """
+        if name in self._data:
+            del self._data[name]
+        else:
+            raise KeyError('{} not in {} object store'
+                           .format(name, self._cim_object_type))
+
+    def iter_names(self):
+        """
+        Overrides the abstract method.
+
+        Returns iterator for names of the objects in the object store
+
+        """
+        return six.iterkeys(self._data)
+
+    def iter_values(self, copy=True):
+        """
+        Overrides the abstract method.
+
+        Returns an iterator for the object values in the object store
+        """
+        for value in six.itervalues(self._data):
+            if copy:
+                yield(value.copy())
+            else:
+                yield(value)
+
+    def len(self):
+        """
+        Overrides the abstract method.
+
+        :class:`~pywbem_mock._baserepository.ObjectStoreAPI.len`
+
+        Returns count of number of objects in the object store dictionary.
+        """
+        return len(self._data)
+
+
+class InMemoryRepository(BaseRepositoryAPI):
+    """
+    Defines the data store and access methods for a simple CIMClass,
+    CIMInstance, and CIMQualifierDeclaration repository that maintains the date
+    in memory.
+
+    The API for this data store is defined in the BaseRepositoryAPI class and
+    the ObjectStoreAPI class.
+    """
+    def __init__(self, initial_namespace=None):
+        """
+        Initialize the InMemoryRepository by creating the top level dictionary.
+        The initial_namespace is optional.  If provided, that namespace
+        is added to the empty repository.
```

Comment:
I think this whole introduction can be removed in the `__init__()` method. It can start right away with the Parameters section. Look at the generated docs.

Reply:
How about just Initialize the InMemoryRepository.  DONE if that is OK

Reply:
Look at the generated docs. It does not combine nicely with the class description.

Reply:
Removed

Reply:
The docstring still contains "Initialize the InMemoryRepository.". If you look at the generated docs, it does not ready very nicely there. I suggest to remove that sentence and to have only the Parameters section in the docstring of `__init__()`.

Reply:
For some reason, this comment does not show up in the files changed view. I have created a new comment and this one here is marked as resolved.